okhttp/src/jvmTest/java/okhttp3/CallTest.kt

```diff
@@ -1022,6 +1022,8 @@ open class CallTest {
     executeSynchronously(request)
       .assertCode(200)
       .assertBody("success!")
+
+    assertThat(client.routeDatabase.failedRoutes.single().proxy.address()).isEqualTo(server.toProxyAddress().address())
```

Comment:
This is the only failing test when `routeDatabase.failed` is not called.  Seems like we are missing coverage.

Reply:
Or perhaps, the coverage we do have (in RouteFailureTest) is overly pessimistic??